### PR TITLE
Add calendar view

### PR DESCRIPTION
**Pull request by pirland**
Added Calendar section to Todos/Home page, and added bootstrap tab functionality to allow switching between sections.

### CODE REVIEW

File: TodoList.Web/Controllers/TodosController.cs [hunk 1]
```diff
@@ -34,15 +34,23 @@ public async Task<IActionResult> Home()
             var currentUser = await _userManager.GetUserAsync(User);
             if (currentUser == null) return Challenge();
 
+            var now = DateTime.Now;
+            var cal = new CalendarViewModel(now.Month, now.Year);
```
Comment (amoraitis) [inline TodoList.Web/Controllers/TodosController.cs:38]: Please give it a better name such as `calendarModel`

File: TodoList.Web/Controllers/TodosController.cs [hunk 2]
```diff
@@ -34,15 +34,23 @@ public async Task<IActionResult> Home()
             var currentUser = await _userManager.GetUserAsync(User);
             if (currentUser == null) return Challenge();
 
+            var now = DateTime.Now;
```
Comment (amoraitis) [inline TodoList.Web/Controllers/TodosController.cs:37]: You can omit that variable and use `DateTime.Now` in each occurrence. It is used only tow times.

File: TodoList.Web/Views/Todos/Home.cshtml [hunk 3]
```diff
@@ -8,131 +8,198 @@
...
-            <td>
-                <a asp-action="Edit" asp-route-id="@item.Id" class="btn btn-secondary">Edit</a>
-                <a asp-action="Details" asp-route-id="@item.Id" class="btn btn-info">Details</a>
-                <a asp-action="Delete" asp-route-id="@item.Id" class="btn btn-danger">Delete</a>
-            </td>
-        </tr>
-        }
-    </tbody>
-</table>
+                    @while (days < Model.CalendarViewModel.NumberOfDays)
```
Comment (amoraitis) [inline TodoList.Web/Views/Todos/Home.cshtml:185]: I don't think that the Monthly todos are being shown in that view.

File: TodoList.Core/Services/TodoItemService.cs [hunk 4]
```diff
@@ -137,6 +137,15 @@ public async Task<IEnumerable<TodoItem>> GetDueTo2DaysItems(ApplicationUser user
                 .ToArrayAsync();
         }
 
+        public async Task<IEnumerable<TodoItem>> GetMonthlyItems(ApplicationUser user, int month)
+        {
+            return await _context.Todos
+                .Where(t => t.UserId == user.Id && !t.Done)
+                 //&& DateTime.Compare(DateTime.UtcNow.AddDays(1), t.DueTo.ToDateTimeUtc()) >= 0)
```
Comment (amoraitis) [inline TodoList.Core/Services/TodoItemService.cs:144]: Do we need this comment?

File: TodoList.Core/Services/TodoItemService.cs [hunk 5]
```diff
@@ -137,6 +137,15 @@ public async Task<IEnumerable<TodoItem>> GetDueTo2DaysItems(ApplicationUser user
                 .ToArrayAsync();
         }
 
+        public async Task<IEnumerable<TodoItem>> GetMonthlyItems(ApplicationUser user, int month)
```
Comment (amoraitis) [inline TodoList.Core/Services/TodoItemService.cs:140]: You can return an `IList<Tuple<TodoItem, TodoItem>>`, the list will have a maximum of `31` and its item of the list will hold 2 items to be shown in the front-end.

https://docs.microsoft.com/en-us/dotnet/api/system.tuple-2?view=netcore-2.2

File: TodoList.Web/Models/CalendarViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TodoList.Web.Models
+{
+    public class CalendarViewModel
```
Comment (amoraitis) [inline TodoList.Web/Models/CalendarViewModel.cs:9]: Can this hold an `IList<Tuple<TodoItem, TodoItem>>` property?

Also, you can use the List's constructor here https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.-ctor?view=netcore-2.2#System_Collections_Generic_List_1__ctor_System_Int32_

File: TodoList.Web/Models/HomeViewModel.cs [hunk 7]
```diff
@@ -7,5 +7,7 @@ public class HomeViewModel
     {
         public IEnumerable<TodoItem> RecentlyAddedTodos { get; set; }
         public IEnumerable<TodoItem> CloseDueToTodos { get; set; }
+        public IEnumerable<TodoItem> MonthlyToTodos { get; set; }
+        public CalendarViewModel CalendarViewModel { get; set; }
```
Comment (amoraitis) [inline TodoList.Web/Models/HomeViewModel.cs:11]: Why those two properties should be different? Couldn't `CalendarViewModel` contain the todo list items for its day?

**Review by amoraitis — CHANGES_REQUESTED** [review]
Please check my comments left on each file.

File: TodoList.Web/Controllers/TodosController.cs (hunk 1 above)
Comment (pirland) [reply TodoList.Web/Controllers/TodosController.cs:38]: Definitely, will do.

File: TodoList.Web/Controllers/TodosController.cs (hunk 2 above)
Comment (pirland) [reply TodoList.Web/Controllers/TodosController.cs:37]: Will do.

File: TodoList.Web/Views/Todos/Home.cshtml (hunk 3 above)
Comment (pirland) [reply TodoList.Web/Views/Todos/Home.cshtml:185]: I seem to have left out the most relevant line.  I have rewritten it slightly to allow for the possibility of multiple todos in one day.  I will push the updated view.

File: TodoList.Core/Services/TodoItemService.cs (hunk 4 above)
Comment (pirland) [reply TodoList.Core/Services/TodoItemService.cs:144]: We do not.  Good catch.

File: TodoList.Core/Services/TodoItemService.cs (hunk 5 above)
Comment (pirland) [reply TodoList.Core/Services/TodoItemService.cs:140]: I'm not completely following here.  Are you talking about returning an IList<Tuple<string, string>>, with the two strings being a field from the Todo entity?

File: TodoList.Web/Models/HomeViewModel.cs (hunk 7 above)
Comment (pirland) [reply TodoList.Web/Models/HomeViewModel.cs:11]: There is not currently a strong functional connection between the two, but I'll move MonthlyTodos into CalendarViewModel in anticipation of a future additions tying them together.

**Comment by amoraitis** [comment]
Closing this and continuing to branch `feature/15-calendar-view`. I will create a new PR.

**Pull request closed without merging**